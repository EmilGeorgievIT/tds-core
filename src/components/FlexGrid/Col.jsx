import React from 'react'
import PropTypes from 'prop-types'

import { Col as ReactFlexboxGridCol } from 'react-flexbox-grid'

import safeRest from '../../utils/safeRest'

const removeProps = ({
  xs,
  sm,
  md,
  lg,
  xl,
  xsOffset,
  smOffset,
  mdOffset,
  lgOffset,
  xlOffset,
  ...rest
}) => safeRest(rest)

const Col = ({ span, offset, children, ...rest }) => (
  <ReactFlexboxGridCol {...removeProps(rest)} xs={span || true} xsOffset={offset}>
    {children}
  </ReactFlexboxGridCol>
)

Col.propTypes = {
  span: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
  offset: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
  children: PropTypes.node.isRequired,
}

Col.defaultProps = {
  span: undefined,
  offset: undefined,
}

export default Col
