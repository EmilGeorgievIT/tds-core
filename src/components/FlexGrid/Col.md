Columns automatically expand to fill the width of its container. You can specify a span amount from 1 to 12.
