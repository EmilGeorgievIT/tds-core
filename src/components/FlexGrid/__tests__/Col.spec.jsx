import React from 'react'
import { shallow } from 'enzyme'

import { Col as ReactFlexboxGridCol } from 'react-flexbox-grid'

import Col from '../Col'

describe('Col', () => {
  const doShallow = (props = {}) => shallow(<Col {...props}>Some content</Col>)

  it('renders', () => {
    const col = doShallow()

    expect(col).toMatchSnapshot()
  })

  it('renders a react-flexbox-grid Col component', () => {
    const col = doShallow()

    expect(col).toMatchSelector(ReactFlexboxGridCol)
  })

  it('has auto width by default', () => {
    const col = doShallow()

    expect(col).toHaveProp('xs', true)
  })

  it('can span a specified number of columns', () => {
    const col = doShallow({ span: 10 })

    expect(col).toHaveProp('xs', 10)
  })

  it('can offset a number of columns', () => {
    const col = doShallow({ offset: 2 })

    expect(col).toHaveProp('xsOffset', 2)
  })

  it('does not support responsive column spans', () => {
    const col = doShallow({ xs: 1, sm: 2, md: 3, lg: 4, xl: 5, span: 10 })

    expect(col).toHaveProp('xs', 10)
    expect(col).not.toHaveProp('sm')
    expect(col).not.toHaveProp('md')
    expect(col).not.toHaveProp('lg')
    expect(col).not.toHaveProp('xl')
  })

  it('does not support responsive offsets', () => {
    const col = doShallow({
      xsOffset: 1,
      smOffset: 2,
      mdOffset: 3,
      lgOffset: 4,
      xlOffset: 5,
      offset: 10,
    })

    expect(col).toHaveProp('xsOffset', 10)
    expect(col).not.toHaveProp('smOffset')
    expect(col).not.toHaveProp('mdOffset')
    expect(col).not.toHaveProp('lgOffset')
    expect(col).not.toHaveProp('xlOffset')
  })

  it('passes additional attributes to the react-flexbox-grid Col', () => {
    const col = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(col).toHaveProp('id', 'the-id')
    expect(col).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const col = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(col).not.toHaveProp('className', 'my-custom-class')
    expect(col).not.toHaveProp('style')
  })
})
