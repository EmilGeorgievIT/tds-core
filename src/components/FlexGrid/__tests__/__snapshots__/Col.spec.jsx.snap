// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col renders 1`] = `
<Col
  xs={true}
>
  Some content
</Col>
`;
